src/resource.c

```diff
@@ -568,24 +568,30 @@ coap_add_resource(coap_context_t *context, coap_resource_t *resource) {
   resource->context = context;
 }
 
+/*
+ * Input context is ignored, but param left there to keep API consistent
+ */
 int
 coap_delete_resource(coap_context_t *context, coap_resource_t *resource) {
-  if (!context || !resource)
+  if (!resource)
     return 0;
 
-  if (resource->is_unknown && (context->unknown_resource == resource)) {
-    coap_free_resource(context->unknown_resource);
-    context->unknown_resource = NULL;
-    return 1;
+  context = resource->context;
+
+  if (resource->is_unknown) {
+    if (context && context->unknown_resource == resource) {
+      context->unknown_resource = NULL;
+    }
   }
-  if (resource->is_proxy_uri && (context->proxy_uri_resource == resource)) {
-    coap_free_resource(context->proxy_uri_resource);
-    context->proxy_uri_resource = NULL;
-    return 1;
+  else if (resource->is_proxy_uri) {
```

Comment:
This PR looks good so far but wrong coding style keeps sneaking in (here the line break before `else`).

Reply:
I guess that I have a bad coding habit, but it is easier on my eyes when scanning the code.
So
````
  if (resource->is_unknown) {
    if (context && context->unknown_resource == resource) {
      context->unknown_resource = NULL;
    }
  }
  else if (resource->is_proxy_uri) {
    if (context && context->proxy_uri_resource == resource) {
      context->proxy_uri_resource = NULL;
    }
  }
  else if (context) {
    /* remove resource from list */
    RESOURCES_DELETE(context->resources, resource);
  }
````
should realy be 
````
  if (resource->is_unknown) {
    if (context && context->unknown_resource == resource) {
      context->unknown_resource = NULL;
    }
  } else if (resource->is_proxy_uri) {
    if (context && context->proxy_uri_resource == resource) {
      context->proxy_uri_resource = NULL;
    }
  } else if (context) {
    /* remove resource from list */
    RESOURCES_DELETE(context->resources, resource);
  }
````
then?

Reply:
Yes.

Reply:
Code updated and pushed.

Reply:
Thanks!